stress: tweak args for `deadlock` and `race` nightlies

rickystewart opened the pull request:
Increase timeouts, reduce the number of test runs, unset
GITHUB_API_TOKEN for nightlies that aren't working fully yet.

Also set `GOTRACEBACK=all` for all EngFlow tests.

Epic: [CRDB-8308](https://cockroachlabs.atlassian.net/browse/CRDB-8308)
Release note: None

rickystewart committed:
stress: tweak args for `deadlock` and `race` nightlies

Increase timeouts, reduce the number of test runs, unset
GITHUB_API_TOKEN for nightlies that aren't working fully yet.

Also set `GOTRACEBACK=all` for all EngFlow tests.

Epic: CRDB-8308
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114681)
<!-- Reviewable:end -->

rickystewart committed:
bazel: `race` does not imply disabling sharding

On the EngFlow cluster, we would like to take advantage of `race`
without disabling sharding, so updating this configuration is sensible.
Anywhere where it's useful to disable sharding, we do it inline.

Epic: CRDB-8308
Release note: None

rickystewart commented:
Ignore the first commit which is from #115180.
